lucene/core/src/java/org/apache/lucene/search/DisjunctionSumScorer.java

```diff
@@ -43,10 +47,25 @@ protected float score(DisiWrapper topList) throws IOException {
     return (float) score;
   }
 
+  @Override
+  public int advanceShallow(int target) throws IOException {
+    int min = DocIdSetIterator.NO_MORE_DOCS;
+    for (Scorer scorer : scorers) {
+      if (scorer.docID() <= target) {
+        min = Math.max(min, scorer.advanceShallow(target));
```

Comment:
I think you meant to do `Math.min` here, not `Math.max`?

```suggestion
        min = Math.min(min, scorer.advanceShallow(target));
```

Reply:
Indeed! Thank you for catching this!